setup_chroot.sh

```diff
@@ -83,12 +83,13 @@ build_chroot()
 	echo -e "\n${COLOR_ON}Creating /etc/schroot/chroot.d/${CHROOT_NAME}.conf...${COLOR_OFF}" 
 	printf "[${CHROOT_NAME}]\ndescription=Ubuntu 12.04 Precise for ${pkg}\ndirectory=${CHROOT_DIR}/${CHROOT_NAME}\npersonality=${personality}\ngroups=sudo\nroot-groups=sudo\npreserve-environment=true\ntype=directory\n" | sudo tee /etc/schroot/chroot.d/${CHROOT_NAME}.conf
 
-	# Add the Ubuntu GPG key to apt
-	sudo apt-key adv --keyserver keyserver.ubuntu.com --recv-keys 0x40976EAF437D05B5
+	# Get the Ubuntu GPG key for Precise packages
+	PRECISE_KEYRING=/tmp/ubuntu-precise-keyring.gpg
+	gpg --keyring=${PRECISE_KEYRING} --no-default-keyring --keyserver keyserver.ubuntu.com --receive-keys 0x40976EAF437D05B5
```

Comment:
I'm a bit concerned that the 64-bit long key ID is within easy reach for collision attacks, so preimage attacks (which is what a malicious keyserver operator or a man-in-the-middle would need to achieve to do something evil here) are not necessarily as far off as we would like.

This is also relying on gpg verifying that the keyserver actually gave us the key we asked for, which it didn't always (although [it has since 2014](https://dev.gnupg.org/T1579).)